controllers/drcluster_controller.go

```diff
@@ -277,6 +272,34 @@ func filterDRClusterMCV(mcv *viewv1beta1.ManagedClusterView) []ctrl.Request {
 	}
 }
 
+func filterDRClusterSecret(reader client.Reader, secret *corev1.Secret) []ctrl.Request {
+	drcusters := &ramen.DRClusterList{}
+	if err := reader.List(context.TODO(), drcusters); err != nil {
+		return []reconcile.Request{}
+	}
+
+	requests := []reconcile.Request{}
+
+	for _, drcluster := range drcusters.Items {
+		s3ProfileName := drcluster.Spec.S3ProfileName
+
+		if s3ProfileName != NoS3StoreAvailable {
+			s3StoreProfile, err := GetRamenConfigS3StoreProfile(context.TODO(), reader, s3ProfileName)
+			if err == nil {
```

Comment:
Our convention is to handle errors first - this ensures that we never ignore errors, and make the code easier to follow by reducing indentation:

    if err != nil {
       continue // maybe also log, why would accessing s3 profile fail?
    }

    process the profile...

Reply:
Done